### PR TITLE
Support JWT authz in Graphql

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java
Comment (olim7t) [inline graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java:134]: When do we hit that branch? The only place where the driver's query builder creates a `TupleTerm` is `QueryBuilder.tuple()`, and it's not referenced anywhere in this project. `DmlFetcher.toCqlTerm` only creates instances of `RawTerm`.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/AllKeyspacesFetcher.java
Comment (olim7t) [inline graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/AllKeyspacesFetcher.java:58]: I think it would be better to push that logic into `KeyspaceFormatter`: check as we go, instead of parsing everything and checking after the fact. It will also allow code reuse between the single-keyspace and multi-keyspace methods.

Comment (olim7t) [inline graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/AllKeyspacesFetcher.java:58]: What is the expected behavior if the user is not authorized for all keyspaces: fail the whole query, or silently omit the restricted ones?

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateTableFetcher.java
Comment (olim7t) [inline graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateTableFetcher.java:50]: These 5 lines are the same for all table operations, only the scope changes. We could extract a `TableFetcher` parent class where `getQuery` performs the check and delegates to an abstract `getQuery(DataFetchingEnvironment, String, String)`.

File: graphqlapi/src/test/java/io/stargate/graphql/schema/GraphQlTestBase.java
Comment (olim7t) [inline graphqlapi/src/test/java/io/stargate/graphql/schema/GraphQlTestBase.java:79]:
```suggestion
      when(authorizationService.authorizedDataRead(actionCaptor.capture(), eq(token), any()))
          .then(i -> actionCaptor.getValue().call());
```

**Review by olim7t — COMMENTED** [review]
A few comments on the GraphQL part.
Depending on the ETA for #413, it might make sense to wait because it should make it easier to retrieve values from the queries.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java:134]: I ran into this for "IN" queries like

```sql
SELECT l FROM collections."PkListTable" WHERE l IN ([1,2,3],[4,5,6])
```

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java:126]: Good point, I'll update. And agreed, that warning is annoying.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/AllKeyspacesFetcher.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/AllKeyspacesFetcher.java:58]: Good question. I should probably pull this up so we can just filter out the ones the user isn't authorized for.

File: authnz/src/main/java/io/stargate/auth/AuthorizationService.java
Comment (dimas-b) [inline authnz/src/main/java/io/stargate/auth/AuthorizationService.java:83]: nit: Since we not longer returning the `ResultSet`, shall we rename to `authorizeDataWrite` (drop the `d`) following the pattern with `authorizeSchemaRead`, etc?

**Comment by dimas-b** [comment]
How does this interact with #407? Graphql API generates CQL... aren't wee going to do atuhz twice?

**Comment by dougwettlaufer** [comment]
> How does this interact with #407? Graphql API generates CQL... aren't wee going to do authz twice?

The token will only be in the custom payload for the CQL path
https://github.com/stargate/stargate/pull/407/files#diff-0d01320b95459a7d0d2a6e36a7d5fb6e4cad3a7019e50e995a426ea24c8e7951R117

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java
Comment (olim7t) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java:133]: Ah my bad, that's indeed how IN relations work.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateTableFetcher.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateTableFetcher.java:50]: Made the change. Let me know what you think.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java
Comment (olim7t) [inline graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java:117]: Nit: this method is used both for tables and UDTs, so
```suggestion
                "Not returning table/UDT {}.{} due to not being authorized",
```
Although this is just a log, so maybe it doesn't matter.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java:52]: SingleKeyspaceFetcher will still call this method, right?

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java:117]: I'm only checking tables right now with the `if(childFieldName.equals("table"))` up above

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java:116]: Yeah I guess I can change to debug. Since it's just filtering them out I felt like there needed to be some sort of signal.

File: graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java
Comment (dougwettlaufer) [reply graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java:52]: Oh, never mind. It goes to the other formatResult.

**Review by dimas-b — APPROVED** [review]
LGTM. One naming comment in `AuthorizationService`

File: authnz/src/main/java/io/stargate/auth/AuthorizationService.java
Comment (dougwettlaufer) [reply authnz/src/main/java/io/stargate/auth/AuthorizationService.java:83]: Sure, will change after I merge into the base branch to avoid waiting on CI.

**Pull request merged**
